### DISCUSSION TITLE
How to customize pedestrian attributes and social force model？

### DISCUSSION THREAD

**Discussion by verse-y** [PedestrianDynamics/jupedsim#1397] [General]
How to customize pedestrian attributes and social force model? For example, I want to add the role of a leader in the environment, ordinary pedestrians will move towards the leader, and the leader has the ability to route by himself. How do I do that?

**Comment by chraibi** [comment]
Hi @verse-y,
I guess what you want to achieve applies for all models and is not specific to the SFM.
I can imagine doing it using the concept of [direct-steering](https://www.jupedsim.org/stable/concepts/routing.html#direct-steering).

Like you can assign the target of everyone else to be the position of the "leader". 

Is that what you mean?

**Reply by verse-y** [reply]
Yes, that's what I want to achieve. I want to study the leader-based evacuation problem, but I do not know how to achieve the leader-follower effect with jupedsim.

**Reply by Ozaq** [reply]
In that case you would assign your leaders a journey as described in the getting started guide. Followers would be configured to use Direct Steering and in your python code you would need to set the target position on the followers to be the current position of your leader

**Reply by chraibi** [reply]
I think that's the right approach. Maybe not the exact current position of the leader, but somewhere nearby. Like within a circle of radius 1 with the leader being at the center.

**Reply by verse-y** [reply]
Thank you for your patience. In this case, does the leader have to be fixed or can be dynamically moved? That is to say, whether the follower's target of Direct Steering can be changed dynamically. And can different types of pedestrians be marked with different colors? For example, the leader is labeled red and the follower is labeled blue.

**Reply by Ozaq** [reply]
You can set the target of direct steering at any position at any iteration

**Comment by chraibi** [comment]
Yes, it is possible to change the coloring according to the role of the agents (leader/followers). 
You can change the anime function by yourself and submit a PR (would be the best) or open an issue. 

Here how the result might look like if you use direct steering

https://github.com/PedestrianDynamics/jupedsim/assets/5772973/bc37418e-7854-4e1c-9104-6ecc3de592c3

**Comment by verse-y** [comment]
I got it, I will try it, thank you for your answer.

**Reply by chraibi** [reply]
@verse-y let us know if you need help with this. There is an open PR #1398 that demonstrates direct steering.
